### ISSUE TITLE
 Arguments to macro show as text in arXiv HTML version of article

### ISSUE THREAD

**Issue by ghenter** [arXiv/html_feedback#258]
### Description

We use a macro to make bold text with the same spacing/kerning as regular text bold, for alignment in tables that mix bold and non-bold numbers. The macro is defined as `\newcommand{\tablebf}[1]{\pdfliteral direct {2 Tr 0.5 w}#1\pdfliteral direct {0 Tr 0 w}}`. Unfortunately, it does not work in the HTML version of arXiv. For example, invoking the macro as `\tablebf{14.8M}` incorrectly shows "2 Tr 0.5 w14.8M0 Tr 0 w" on the HTML page, whereas the PDF correctly shows only the text "14.8M" in bold.

### (Optional:) Please add any files, screenshots, or other information here.

Screenshot from Overleaf of our exact macro code, for reference:
<img width="523" alt="image" src="https://github.com/arXiv/html_feedback/assets/9961926/518f4222-f390-46d8-9e17-6ce6ded184e1">

Screenshot from HTML version of article, showing the error:
<img width="213" alt="image" src="https://github.com/arXiv/html_feedback/assets/9961926/e7a6836f-b132-4af1-b7f9-6b553a98650d">

Screenshot from rendered PDF, showing the expected result:
<img width="141" alt="image" src="https://github.com/arXiv/html_feedback/assets/9961926/ea1903e2-6389-49c8-9371-8b810504160d">

### (Required) What is this issue most closely related to? Select one.

Other

### Internal issue ID

Since this field was not set automatically, I am entering text so that "Submit new issue" can be clicked.

### Paper URL

Since this field was not set automatically, I am entering text so that "Submit new issue" can be clicked.

### Browser

Since this field was not set automatically, I am entering text so that "Submit new issue" can be clicked.

### Device Type

Since this field was not set automatically, I am entering text so that "Submit new issue" can be clicked.

**Comment by dginev** [comment]
Currently latexml tries to [simply ignore](https://github.com/brucemiller/LaTeXML/blob/6ee12be9311b90280fdd89f5bdb4fa13ca118785/lib/LaTeXML/Package/pdfTeX.pool.ltxml#L199C1-L199C1) `\pdfliteral` directives.

We could enhance latexml to recognize this case, but you're dangerously close to being in PDF-only territory. 

This kind of macro may benefit from the `\iflatexml` conditional when it becomes available, to use the simpler bold directives already supported by latexml in the HTML context, while still using the PDF-optimized variant for PDF.

**Comment by davidcarlisle** [comment]
a more portable and idiomatic latex way to get unextended bold would be `\fontseries{b}\selectfont` (rather than `bx`) see for example

https://tex.stackexchange.com/questions/410739/align-numbers-based-on-the-decimal-point-with-textbf-and-stars-in-the-table/410741#410741

that isn't to say latexml shouldn't try to handle the pdftex specific markup that already exists in existing documents but simply ignoring it would be acceptable here I think.

**Comment by ghenter** [comment]
Thanks for these insightful and helpful responses! I will look into the method suggested by @davidcarlisle for the future.

Also, apologies if this was a known WONTFIX/duplicate; the arXiv HTML preview encouraged me to submit an error report if I spotted any issues, so that's what I did. :)

I am closing the issue, but you may reopen it if you deem it relevant.

**Comment by davidcarlisle** [comment]
up to the latexml team but I wouldn't have closed this. In general latexml (or any convertor) is going to have to handle all kinds of tex markup, so just because there may be better markup that could have been used doesn't mean that the system should do something reasonable with whatever markup is actually there.

**Comment by dginev** [comment]
I agree the issue should have staid open: this is just some friendly exploratory discussion on our part @ghenter .

I don't have sufficient permissions to reopen, but I know who to ask...

**Comment by chrisjcameron** [comment]
Thanks for the discussion. This project has an ambitious goal and we know it will be more of a marathon than a sprint. I really appreciate the feedback from authors and readers -- it helps us identify and prioritize issues.

**Comment by u-fischer** [comment]
You could use the pdfrender package instead of the  primitive. This has the benefit that it would support also lualatex (and perhaps later other engines) and would probably allow latexml to provide an alternative:

~~~~
\documentclass{book}
\usepackage{pdfrender}
\newcommand\tablebf[1]{\textpdfrender{TextRenderingMode=FillStroke,LineWidth=0.5}{#1}}
\begin{document}
blub \tablebf{some text} more text
\end{document}
~~~~

This inserts the same literals into the PDF:

~~~~
2 Tr
0.5 w
 [-333(some)-334(text)]TJ
0 Tr
1 w
~~~~

and gives this as output

![image](https://github.com/arXiv/html_feedback/assets/4047173/84493020-6eab-47fa-9b8d-fb5232a12889)

**Comment by dginev** [comment]
Resolved, latexml should now be content-preserving on that formulation of `\pdfliteral`. We have not emulated the embellishments, just rescued the author-provided argument.
